src/main/java/org/testng/xml/TestNGContentHandler.java

```diff
@@ -90,8 +90,7 @@ public TestNGContentHandler(String fileName, boolean loadClasses) {
   public InputSource resolveEntity(String systemId, String publicId)
       throws IOException, SAXException {
     InputSource result;
-    if (Parser.DEPRECATED_TESTNG_DTD_URL.equals(publicId)
-        || Parser.TESTNG_DTD_URL.equals(publicId)) {
+    if (Parser.HTTPS_TESTNG_DTD_URL.equals(publicId)) {
```

Comment:
I 100% do not recommend disabling this logic for the depricated URL.
Doing so will cause users who update TestNG and don't update their xml to be vulnerable to XXE via a MITM.

Reply:
I confirm we need still support previous urls because we don't know where they come from (could be proxy cache or something else configured in the xml catalog)

Reply:
Also, this should have support for all 4 possible permutations of the url. That way if someone uses the old depricated DTD but over HTTPS everything will still work.

Reply:
@juherr - The only reason why I chose to remove it was because its named **Deprecated URL** . The question is... should it still remain there ?

Reply:
Irrespective of that question I have pushed in the changes to take care of the comments. @juherr  Please help take a look.